Backend/services/userService.js

```diff
@@ -0,0 +1,51 @@
+const mongoose = require("mongoose");
+const User = require("../models/userModel");
+const hashPassword = require("../utils/hashPassword");
+
+const checkExistingUser = async (email) => {
+  return await User.findOne({ email });
+};
+
+const createUser = async (userData) => {
+  const { userName, email, password } = userData;
+
+
+  const user = new User({
+    userName,
+    email,
+    password: password,
+  });
+
+  return await user.save();
+};
+
+const findUserByEmail = async (email) => {
+  return await User.findOne({ email }).exec();
+};
+
+const updatePassword = async (user, newPassword) => {
+  user.password = newPassword;
+  user.resetToken = null;
+
+  return await user.save();
+};
+
+const findUserByResetToken = async (resetPasswordToken) => {
```

Comment:
Threre is no need to create three seprate functions (findUserByEmail, checkExistingUser, findUserByResetToken) for finding a single user.
You can create a single service to findOne User based on a specific condition provided when calling this service.